docs/run_test/understand_results.md

```diff
@@ -0,0 +1,71 @@
+# Understand test results
+
+- [Overview](#overview)
+- [Intermediate results](#intermediate-results)
+- [Final results](#final-results)
+
+## Overview
+
+It's essential to understand the results after running tests. LISA has 7 kinds
+of test results in total: 3 of which are intermediate results, and 4 of which
+are final results, as explained in sections below. Each test case can and will
+be moved from one result to another but can never have two or more results at
+the same time. 
+
+## Intermediate results
+
+An intermediate result shows information of an unfinished test. It will show up
+when a test changes its state. If a test terminates because of error or
+exception, only the intermediate result will be provided.
+
+- **QUEUED**
+
+  QUEUED tests are tests that are created, and planned to run (but have not
+  started yet). They suggest that there are some tests waiting to be performed.
+
+  QUEUED tests will move forward to ASSIGNED.
+
+- **ASSIGNED**
+
+  ASSIGNED tests are tests that are assigned to an environment, and will start
+  to run once the environment is deployed. They suggest some environmental
+  setting up is going on. 
+
+  ASSIGNED tests will end with FAILED if the environment fails to deploy. They
+  will also move backward to QUEUED again if the environment does not fit the
+  requirement of the test.
```

Comment:
Do we ever need that change of status in the first place, given env reqs don't change dynamically, AFAIK, and the check is simple enough? (I mean back from ASSIGNED)

Reply:
Yes, The reason of case move back to Queued is due to all cases need to be reselected. the described condition wont' happen (the env doesn't fit the test).

Reply:
We should/could also state that queued also means a pre-selection has already taken place (modulo environment), and the test has passed it (runbook criteria), no?

Reply:
@WenyiQ 
Yes, we can specify the queued case is selected by criteria. If a case is not queued, check the log if the case is included by extension paths, and criteria conditions.